python/ray/tune/trial_executor.py

```diff
@@ -31,15 +31,14 @@ def has_resources(self, resources):
         raise NotImplementedError("Subclasses of TrialExecutor must provide "
                                   "has_resources() method")
 
-    def start_trial(self, trial, checkpoint=None):
-        """Starts the trial restoring from checkpoint if checkpoint != None.
-
-        If an error is encountered when starting the trial, an exception will
-        be thrown.
+    def start_trial(self, trial, checkpoint=None, raise_on_failure=False):
```

Comment:
Why add raise_on_failure? It seems to complicate the start trial path more.

We should also avoid changing the interface here since it is public for other execution backends.